src/index.ts

```diff
@@ -248,9 +259,46 @@ export interface CanonicalPurpose {
   purposeCodes?: string[]
 }
 
+// IdentityLocation is the location on the page from which to retrieve identity information
+export enum IdentityType {
+  IDENTITY_TYPE_UNDEFINED = "",
+  IDENTITY_TYPE_DATA_LAYER = "dataLayer",
+  IDENTITY_TYPE_WINDOW = "window",
+  IDENTITY_TYPE_COOKIE = "cookie",
+  IDENTITY_TYPE_MANAGED = "managedCookie", // this is created if necessary and stored in a cookie with the associated name
+  IDENTITY_TYPE_LOCAL_STORAGE = "localStorage",
+  IDENTITY_TYPE_SESSION_STORAGE = "sessionStorage",
+  IDENTITY_TYPE_QUERY_STRING = "queryString",
+}
+
+// IdentityFormat is the encoding of the string identity value
+export enum IdentityFormat {
+  IDENTITY_FORMAT_UNDEFINED = "",
+  IDENTITY_FORMAT_STRING = "string",
+  IDENTITY_FORMAT_JSON = "json",
+  IDENTITY_FORMAT_JWT = "jwt",
+  IDENTITY_FORMAT_QUERY = "query",
+  IDENTITY_FORMAT_SEMICOLON = "semicolon",
+}
+
+// Identity represents all the metadata for an identifier on the page
 export interface Identity {
-  type: string
-  variable: string
+
+  // type is the location on the page from which to retrieve identity information
+  type: IdentityType
```

Comment:
@sethyates think the string equality checking we do now will be compatible with this